### PR TITLE
10.0 mig medical appointment (Requires #80 )

### PR DESCRIPTION
**Pull request by jcdrubay**
Requires #80

### CODE REVIEW

File: medical_appointment/README.rst
Comment (lasley) [inline medical_appointment/README.rst:16]: s/insall/install

Comment (lasley) [inline medical_appointment/README.rst:16]: May want to use a bullet instead of a `.` too. So a `-` or `*`

File: medical_appointment/README.rst
Comment (lasley) [inline medical_appointment/README.rst:23]: s/./*

File: medical_appointment/__manifest__.py
Comment (lasley) [inline medical_appointment/__manifest__.py:25]: s/licence/license

File: medical_appointment/data/medical_appointment_sequence.xml
Comment (lasley) [inline medical_appointment/data/medical_appointment_sequence.xml:3]: It's best to make these explicit lines. Also, please drop my `-TODAY` - that's BS. We can make is 2015-2016 for mine, and 2016 for yours (or maybe 2017, depending on when you wrote this).

Doing this allows the copyrights to be maintained individually as time progresses, instead of our two copyrights being bound to each other. Minor nuances, but good practice 😉

File: medical_appointment/data/medical_appointment_sequence.xml
Comment (lasley) [inline medical_appointment/data/medical_appointment_sequence.xml:7]: `<odoo noupdate="1">` & can remove data tag

File: medical_appointment/models/medical_appointment.py
Comment (lasley) [inline medical_appointment/models/medical_appointment.py:2]: Same note regarding copyright. Also, it's 2017 now 😉

File: medical_appointment/models/medical_appointment.py
Comment (lasley) [inline medical_appointment/models/medical_appointment.py:7]: Alphabetize imports

File: medical_appointment/views/medical_appointment_view.xml
Comment (lasley) [inline medical_appointment/views/medical_appointment_view.xml:220]: Please move this to roadmap. Commented code sucks for maintenance

**Review by lasley — CHANGES_REQUESTED** [review]
Thanks @jcdrubay - few comments

File: medical_appointment/models/medical_appointment.py
Comment (jcdrubay) [reply medical_appointment/models/medical_appointment.py:7]: Following this logic: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports

odoo comes 3rd!

File: medical_appointment/data/medical_appointment_sequence.xml
Comment (jcdrubay) [reply medical_appointment/data/medical_appointment_sequence.xml:7]: I did not know this worked.
I ll test, but if it works we should recommend an update here: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#format

File: medical_appointment/models/medical_appointment.py
Comment (jcdrubay) [reply medical_appointment/models/medical_appointment.py:2]: Would this be good?

```
# -*- coding: utf-8 -*-
# Copyrights
#    2016-2017 LasLabs Inc.
#    2017 Komit
# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```

File: medical_appointment/data/medical_appointment_sequence.xml
Comment (jcdrubay) [reply medical_appointment/data/medical_appointment_sequence.xml:3]: Like that, right?
```
<!-- Copyrights
        2015-2017 LasLabs [https://laslabs.com]
        2017 Komit [komit-consulting.com]
     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
```

File: medical_appointment/views/medical_appointment_view.xml
Comment (jcdrubay) [reply medical_appointment/views/medical_appointment_view.xml:220]: I removed it completely, I don't think we need it in generic. It would be annoying in case there is a receptionist...

Comment (jcdrubay) [reply medical_appointment/views/medical_appointment_view.xml:220]: It was this domain when opening the list of appointments.
```
        <field name="domain">['|',('patient_id.user_id.id','=',uid),('doctor.user_id.id','=',uid)]</field>
```

Comment (lasley) [reply medical_appointment/views/medical_appointment_view.xml:220]: Agreed!

File: medical_appointment/models/medical_appointment.py
Comment (lasley) [reply medical_appointment/models/medical_appointment.py:7]: Yup, but the internal ordering of that should be `_, api, exceptions, fields, models`

File: medical_appointment/models/medical_appointment.py
Comment (lasley) [reply medical_appointment/models/medical_appointment.py:2]: The copyright isn't binding unless it's prefaced with `Copyright` (at least in US), so I'm not sure how combining under the one would work TBH. This is probably the best way:
```
Copyright 2016 LasLabs Inc.
Copyright 2017 Komit
```

Note that I don't have a copyright in 2017 - I didn't add any code during that year at this point, only you have. That same reasoning is why I said to get rid of my indefinite copyright on the one `2015-TODAY`.

File: medical_appointment/data/medical_appointment_sequence.xml
Comment (lasley) [reply medical_appointment/data/medical_appointment_sequence.xml:7]: Good idea - I'll submit a patch to the template module today

Comment (lasley) [reply medical_appointment/data/medical_appointment_sequence.xml:7]: OCA/maintainer-tools#253. I notice you pointed to the contribution docs and not the template module though. What was your thought on how to change that? Or did you just want to submit the PR yourself?

Comment (jcdrubay) [reply medical_appointment/data/medical_appointment_sequence.xml:7]: I wanted to refer to this line:
`The tag <data> is only used to set not-updatable data with noupdate=1`

I can submit the PR if you like, but it seems that you are already deeply engaged on the topic (https://github.com/OCA/maintainer-tools/pull/253) :)

Comment (jcdrubay) [reply medical_appointment/data/medical_appointment_sequence.xml:7]: https://github.com/OCA/maintainer-tools/pull/255

**Comment by lasley** [comment]
@obulkin - Can you verify that any changes @jcdrubay (or PR suggestions) made in this PR are integrated into #171 please?

**Comment by obulkin** [comment]
It seemed off-topic to include this work in #171 so my current plan is to wait for #174 to be merged and then to review this PR, looking for elements that are still relevant.

**Comment by obulkin** [comment]
I'm closing this PR since ``medical_base_history`` was deprecated in https://github.com/LasLabs/vertical-medical/pull/171 and the ``medical_appointment`` changes suggested here are effectively already in ``release/10.0``.

Thanks for your work on this, @jcdrubay.

**Comment by jcdrubay** [comment]
It has been a great pleasure to contribute something, and I hope that in the future I can contribute "much" more ... :)

**Pull request closed without merging**
